### PR TITLE
Potential v2.3.6 release candidate

### PR DESCRIPTION
**Pull request by AtnNn**
Cherry-pick a few fixes from `next` and add release notes draft.

- [x] I have read and agreed to the RethinkDB Contributor License Agreement

### CODE REVIEW

**Comment by srh** [comment]
Looks good to me.

File: NOTES.md [hunk 1]
```diff
@@ -1,3 +1,44 @@
...
+salvation} of RethinkDB. To get involved in the transition and to see
+how it might affect you, read our blog post on {revelation}.
+
+### Compatibility ###
+
+On 32-bit platforms and on Windows (64 and 32 bit), RethinkDB 2.3.6 servers should not
+be mixed with servers running RethinkDB 2.3.3 or older in the same cluster. Doing so can lead to
+server crashes when using the web UI or when accessing the `logs` system table.
+
+On 64-bit platforms, RethinkDB 2.3.5 servers can be mixed with older RethinkDB 2.3.x
```
Comment (srh) [inline NOTES.md]: Should be "RethinkDB 2.3.6 servers" on line 21 here.  Right?

File: NOTES.md [hunk 2]
```diff
@@ -1,3 +1,44 @@
...
+older version.
+
+### Changes ###
+
+* Server
+ * Improved the compatibility of the web UI with Chrome 48 and Edge (#5878, #5426, #5300) -- @danielmewes
+ * Fixed a crash caused by unwanted connections (#6084) -- @danielmewes
+ * Fixed a crash caused by recreating indexes with active changefeeds (#6093) -- @danielmewes
+ * Sizes passed to `sample` are now bound by the array size limit (#6148) -- @AtnNn
+ * Fixed the wrong behaviour of `match` on empty patterns (#6139) -- @AtnNn
```
Comment (marshall007) [inline NOTES.md]: This description is for the wrong issue. Should be something along the lines of `don't persist stale env_t* in ordered unions`.

I think the empty `match` fix was also slated for `v2.3.6` so I think we need to link that in as well.

**Comment by marshall007** [comment]
@AtnNn it also appears that #6229 was cherry picked into `v2.3.x`. I don't see that mentioned in the release notes either.

**Comment by AtnNn** [comment]
Thanks for the comments. The new draft is ready for review @marshall007 @srh

**Comment by marshall007** [comment]
@AtnNn still missing #6202 I think.

**Comment by marshall007** [comment]
@AtnNn just kidding you beat me to it by 2 minutes :)

**Pull request merged**